docs/specification-ligo.md

```diff
@@ -605,14 +607,13 @@ Parameter (in Michelson):
 ```
 
 - The proposal is saved under `BLAKE2b` hash of proposal value and sender.
-- The `Natural` value: `proposalTokenAmount` determines how many sender's tokens will be frozen in addition to the [fee](#set_fixed_fee_in_token).
-- Sender MUST have enough unfrozen tokens (i. e. `≥ proposalTokenAmount + fee`).
-- Fails with `PROPOSAL_INSUFFICIENT_BALANCE` if the unfrozen token balance of the SENDER
+- The `Natural` value: `proposalTokenAmount` determines how many sender's frozen tokens will be staked in addition to the [fee](#set_fixed_fee_in_token).
+- Sender MUST have enough frozen tokens (i. e. `≥ proposalTokenAmount + fee`) that are not already staked for a proposal or a vote.
+- Fails with `NOT_ENOUGH_FROZEN_TOKENS` if the unstaked frozen token balance of the SENDER
   is less than `proposalTokenAmount + fee`.
 - Fails with `FAIL_PROPOSAL_CHECK` if the proposal is rejected by `proposalCheck` from the configuration.
```

Comment:
Note: we need to add a line about the proposal stage requirement to the propose entrypoint description.

```suggestion
- Fails with `NOT_PROPOSING_PERIOD` if the current period is not a proposing stage.
- Fails with `FAIL_PROPOSAL_CHECK` if the proposal is rejected by `proposalCheck` from the configuration.
```